pkg/pillar/zedcloud/zedcloudmetric.go

```diff
@@ -29,24 +29,30 @@ func maybeInit(ifname string) {
 	}
 }
 
-func ZedCloudFailure(ifname string, url string, reqLen int64, respLen int64) {
+func ZedCloudFailure(ifname string, url string, reqLen int64, respLen int64, authenFail bool) {
 	mutex.Lock()
 	maybeInit(ifname)
 	m := metrics[ifname]
-	m.FailureCount += 1
-	m.LastFailure = time.Now()
-	var u types.UrlcloudMetrics
-	var ok bool
-	if u, ok = m.URLCounters[url]; !ok {
-		u = types.UrlcloudMetrics{}
-	}
-	u.TryMsgCount += 1
-	u.TryByteCount += reqLen
-	if respLen != 0 {
-		u.RecvMsgCount += 1
-		u.RecvByteCount += respLen
+	// if we have authen verify failure, the network part is success
```

Comment:
Not entirely true. If eth0 goes through a proxy and gets some authentication failure and eth1 (which may or may not use a (different) proxy) does not have an authentication failure, then eth1 is clearly broken.

So I think we should continue to count all of the authentication failures as failures, and in addition (next sprint) have a counter for each status enum value for the authentication issues.

Reply:
will pick up this more in next sprint.

Reply:
OK
In addition, I think if you pass the SenderStatus instead of a bool as a last argument then you can easily have a counter per type of failure.